geomesa-cassandra/geomesa-cassandra-datastore/src/test/scala/org/locationtech/geomesa/cassandra/data/CassandraDataStoreTest.scala

```diff
@@ -23,52 +22,53 @@ import org.locationtech.geomesa.features.ScalaSimpleFeature
 import org.locationtech.geomesa.index.geotools.GeoMesaDataStoreFactory.LooseBBoxParam
 import org.locationtech.geomesa.index.utils.{ExplainString, Explainer}
 import org.locationtech.geomesa.utils.collection.SelfClosingIterator
-import org.locationtech.geomesa.utils.conf.GeoMesaSystemProperties.SystemProperty
 import org.locationtech.geomesa.utils.geotools.{SchemaBuilder, SimpleFeatureTypes}
-import org.locationtech.geomesa.utils.io.PathUtils
 import org.specs2.matcher.MatchResult
 import org.specs2.mutable.Specification
 import org.specs2.runner.JUnitRunner
-
-import java.nio.file.{Files, Path}
+import org.specs2.specification.BeforeAfterAll
+import org.testcontainers.containers.GenericContainer
 
 @RunWith(classOf[JUnitRunner])
-class CassandraDataStoreTest extends Specification {
+class CassandraDataStoreTest extends Specification with BeforeAfterAll {
 
   import scala.collection.JavaConverters._
 
   sequential
 
-  var storage: Path = _
   var params: Map[String, String] = _
-  var ds: CassandraDataStore = _
 
-  step {
-    storage = Files.createTempDirectory("cassandra")
+  private val cassandraContainer = new GenericContainer[Nothing]("cassandra:4.0.8")
+  private val scyllaContainer = new GenericContainer[Nothing]("scylladb/scylla:6.2")
+  private var dsList: List[CassandraDataStore] = List()
 
-    System.setProperty("cassandra.storagedir", storage.toString)
+  def beforeAll(): Unit = {
+    cassandraContainer.start()
+    scyllaContainer.start()
```

Comment:
tests are failing to connect to the container - i think you'd have to add exposed ports for 9042 here, and then use `container.getMappedPort(9042)` in the data store params

Reply:
i've migrated from Generic container where we need to handle such things explicitly to CassandraContainer and ScyllaDBContainer, which handles such things out of the box. Could you pls retry the tests?